src/stringProcessing/checkNofollow.js

```diff
@@ -1,17 +1,38 @@
 /** @module stringProcessing/checkNofollow */
 
+// We use an external library, which can be found here: https://github.com/fb55/htmlparser2.
+let htmlparser = require( "htmlparser2" );
+let isString = require( "lodash/isString" );
+
 /**
  * Checks if a links has a nofollow attribute. If it has, returns Nofollow, otherwise Dofollow.
  *
  * @param {string} text The text to check against.
  * @returns {string} Returns Dofollow or Nofollow.
  */
 module.exports = function( text ) {
-	var linkFollow = "Dofollow";
+	let linkFollow = "Dofollow";
+
+	let parser = new htmlparser.Parser( {
+		/**
+		 * Detects if there is a `nofollow` argument in the `rel` attribute of a link.
+		 *
+		 * @param {string} tagName The tag name.
+		 * @param {object} nodeValue The attribute with the keys and values of the tag.
+		 * @returns {void}
+		 */
+		onopentag: function( tagName, nodeValue ) {
+			if ( tagName.toLowerCase() !== "a" || ! isString( nodeValue.rel ) ) {
```

Comment:
the passed anchor text already has tag names and attributes names lower cased because the parser is already used for the text passed to the Paper, and it already lower cases tag names, see `src/stringProcessing/htmlParser.js`. Regardless, the parser parses again the link here and by default it returns lower case tag names. See the docs:
https://github.com/fb55/htmlparser2/wiki/Parser-options

Option: xmlMode
Default: false.

Option: lowerCaseTags
If set to true, all tags will be lowercased. **If xmlMode is disabled, this defaults to true**.

Reply:
If it's not a string (or empty) there is no way `nofollow` would be present in the argument, which is the intended behaviour of this specific implementation.

The alternative is to do the following check: `! ( "rel" in nodeValue )`
Which is much harder to read in my opinion.
And would be spread out over multiple lines per coding standards.

Alternatively `_isUndefined ` could be used from Lodash, but that is also more general and does not remove the need for an import.
I don't think we need to optimize for use of imports as these should be generally available, and if used somewhere else in the code it is justified to use them "everywhere". Could you confirm @atimmer

Reply:
`isString` is fine here, as this ensures that member methods are available to use on the string. It prevents errors such as 'calling method on undefined'.